src/builtins/set.zig

```diff
@@ -63,21 +63,54 @@ const OptState = union(OOptions) {
     vi: ?bool,
     xtrace: ?bool,
 };
-const KnOptions = std.ArrayList(OptState);
+
+pub const KnOptions = struct {
+    allexport: ?bool,
+    errexit: ?bool,
+    ignoreeof: ?bool,
+    monitor: ?bool,
+    noclobber: ?bool,
+    noglob: ?bool,
+    noexec: ?bool,
+    nolog: ?bool,
+    notify: ?bool,
+    nounset: ?bool,
+    verbose: ?bool,
+    vi: ?bool,
+    xtrace: ?bool,
+
+    pub fn init() KnOptions {
+        return KnOptions {
+            .allexport = false,
```

Comment:
if we're gonna init everything to false, and not `null`, we should probably change the type `?bool` -> `bool` right?

or maybe `?bool` => `bool = false` then we can use `.{}`